apps/api/src/lib/id-generation/interfaces/id-repository.interface.ts

```diff
@@ -0,0 +1,10 @@
+import { Maybe } from '../../types/maybe';
+import { UuidDocument } from '../types/UuidDocument';
+
+export interface IIdRepository<TIdFormat> {
```

Comment:
I would comment these methods heavily to explain that create will create a new ID that is available and reserve will update an existing id from available to in use.

There's some overlap between the ID-Repository and the Id-Manager-Service. What additional benefits does the `IdManager` abstraction give us to justify its maintenance cost?

Reply:
The abstraction allows us to inject a dependency whose API is centred around the use cases. This makes testing easier. It also makes command-handler code more declarative.